File: src/algorithm1/day3/MoveZeros.java
package algorithm1.day3;

public class MoveZeros {
    public static void main(String[] args) {
        int[] nums = {1, 0};
        moveZeros(nums);

        for (int num : nums) {
            System.out.println("num = " + num);
        }
    }

    public static void moveZeros(int[] nums) {
        if (nums == null || nums.length == 1) {
            return;
        }

        int zeroPointer = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != 0) {
                int temp = nums[zeroPointer];
                nums[zeroPointer] = nums[i];
                nums[i] = temp;
                zeroPointer++;
            }
        }
    }
}

File: src/algorithm1/kotlinsource/day3/MoveZeros.kt
package algorithm1.kotlinsource.day3

class MoveZeros {
    fun moveZeros(nums: IntArray): Unit {
        var zeroPointer = 0

        for ((index, _) in nums.withIndex()) {
            if (nums[index] != 0) {
                var temp = nums[zeroPointer]
                nums[zeroPointer] = nums[index]
                nums[index] = temp
                zeroPointer += 1
            }
        }
    }
}

fun main() {
    val moveZeros = MoveZeros()
    val nums: IntArray = intArrayOf(0, 1, 0, 3, 12)
    moveZeros.moveZeros(nums)


}